submarine-server/server-submitter/submitter-yarnservice/src/main/java/org/apache/submarine/server/submitter/yarnservice/tensorflow/TensorFlowConfigEnvGenerator.java

```diff
@@ -0,0 +1,104 @@
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import com.fasterxml.jackson.module.jaxb.JaxbAnnotationIntrospector;
+import org.apache.submarine.commons.runtime.conf.Envs;
+import org.apache.submarine.server.submitter.yarnservice.YarnServiceUtils;
+
+public class TensorFlowConfigEnvGenerator {
+
+  private static final ObjectMapper OBJECT_MAPPER = createObjectMapper();
+
+  private static ObjectMapper createObjectMapper() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.setAnnotationIntrospector(
+        new JaxbAnnotationIntrospector(TypeFactory.defaultInstance()));
+    mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    mapper.configure(SerializationFeature.FLUSH_AFTER_WRITE_VALUE, false);
+    mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+    return mapper;
+  }
+
+  public static String getTFConfigEnv(String componentName, int nWorkers,
+      int nPs, String serviceName, String userName, String domain) {
+    String commonEndpointSuffix = YarnServiceUtils
+        .getDNSNameCommonSuffix(serviceName, userName, domain, 8000);
+
+    TFConfigEnv tfConfigEnv =
+        new TFConfigEnv(nWorkers, nPs, componentName, commonEndpointSuffix);
+    return tfConfigEnv.toJson();
+  }
+
+  private static class TFConfigEnv {
+    private final int nWorkers;
+    private final int nPS;
+    private final String componentName;
+    private final String endpointSuffix;
+
+    TFConfigEnv(int nWorkers, int nPS, String componentName,
+          String endpointSuffix) {
+      this.nWorkers = nWorkers;
+      this.nPS = nPS;
+      this.componentName = componentName;
+      this.endpointSuffix = endpointSuffix;
+    }
+
+    String toJson() {
+      ObjectNode rootNode = OBJECT_MAPPER.createObjectNode();
+
+      ObjectNode cluster = rootNode.putObject("cluster");
+      createComponentArray(cluster, "master", 1);
+      createComponentArray(cluster, "worker", nWorkers - 1);
+      createComponentArray(cluster, "ps", nPS);
+
+      ObjectNode task = rootNode.putObject("task");
+      task.put("type", componentName);
+      task.put("index", "$" + Envs.TASK_INDEX_ENV);
+      task.put("environment", "cloud");
+      try {
+        return OBJECT_MAPPER.writeValueAsString(rootNode);
+      } catch (JsonProcessingException e) {
+        throw new RuntimeException("Failed to serialize TF config env JSON!",
+            e);
+      }
+    }
+
+    private void createComponentArray(ObjectNode cluster, String name,
+          int count) {
+      ArrayNode array = cluster.putArray(name);
+      for (int i = 0; i < count; i++) {
+        String componentValue = String.format("%s-%d%s", name, i,
+            endpointSuffix);
+        array.add(componentValue);
+      }
+    }
+  }
+
+
```

Comment:
redundant blank line

Reply:
Thanks for the review! Removed the blank line.